docs/proposals/buildstrategy-annotation-propagation.md

```diff
@@ -0,0 +1,165 @@
+
+- The `build.build.dev/build-run-deletion` annotation on builds controls whether its `BuildRun`s are deleted when the build is deleted.
+- The `build.build.dev/referenced.secret` annotation tells the build controller that a secret is related to builds.
+
+Based on that naming, the other three prefixes reserved by our controllers are: `buildstrategy.build.dev`, `clusterbuildstrategy.build.dev` and `buildrun.build.dev`.
+
+When generating a Tekton `TaskRun`, the idea is to look at the annotations of the `BuildStrategy` or `ClusterBuildStrategy` and copy all annotations over to the `TaskRun`, except those that use one of our own prefixes because we today have no feature anyway where we look for one of our annotations on a `TaskRun` or `Pod`, and except the `kubectl.kubernetes.io/last-applied-configuration` annotation.
+
+Tekton automatically copies all `TaskRun` annotations to the `Pod`, see [pod.go](https://github.com/tektoncd/pipeline/blob/v0.18.1/pkg/pod/pod.go#L256).
+
+For example, this metadata of a cluster build strategy:
+
+```yaml
+apiVersion: build.dev/v1alpha1
+kind: ClusterBuildStrategy
+metadata:
+  annotations:
+    kubernetes.io/egress-bandwidth: 100M
+    clusterbuildstrategy.build.dev/dummy: aValue
+```
+
+will lead to the following metadata on the `TaskRun` (and `Pod`):
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  annotations:
+    kubernetes.io/egress-bandwidth: 100M
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+The implementation requires the [BuilderStrategy interface](../../pkg/apis/build/v1alpha1/buildstrategy.go) to be extended with a `GetAnnotations` functions that is implemented in the [BuildStrategy](../../pkg/apis/build/v1alpha1/buildstrategy_types.go) and [ClusterBuildStrategy](../../pkg/apis/build/v1alpha1/clusterbuildstrategy_types.go) types by returning the object's annotations.
+
+The assignment of the `TaskRun` annotations needs to be done in the [generate_taskrun.go](../../pkg/controller/buildrun/generate_taskrun.go) file in the `GenerateTaskRun` function. The annotations from the build strategy need to be copied to the `TaskRun` except those using one of the four Shipwright Build owned prefixes mentioned under [Proposal](#proposal), and except the `kubectl.kubernetes.io/last-applied-configuration` annotation.
+
+### Risks and Mitigations
+
+A risk is that the build strategy administrator starts to use an annotation-controlled feature that the Kubernetes administrator does not want to be used. If we assess this risk to be high enough, then option (2) from the [alternatives](#alternatives) might be required.
```

Comment:
In Tekton, we are advising use of third party policy engines like OPA to catch such things.  With OPA, the K8S administrator can write policies to say filter which namespaces's pods can have a given k8s annotation.

Use of third party policy engines also seems to be the current leaning of KEP 5 with k8s sig-auth (though certainly there is some amount of fluidity there).

That said, IMO I believe shipwright can likewise defer to the burgeoning k8s policy engine ecosystem here.  And thus your EP should be able to move forward at a high level with your current approach.

Good risk citation though ;-) ....  some reference to how to mitigate this risk could be warranted as part of checking off `User-facing documentation is created in...` in the sign off checklist. 

@adambkaplan @otaviof @sbose78  FYI ^^

Reply:
Good hint @gabemontero. I changed

> If we assess this risk to be high enough, then option (2) from the [alternatives](#alternatives) might be required.

to

> Third-party policy engines like [Open Policy Agent](https://www.openpolicyagent.org/) can be used by the Kubernetes administrator to prevent this without us requiring anything. If this is considered not enough, then option (2) from the [alternatives](#alternatives) might be required.

In Tekton, I was finding this "only" referenced in a few issues. If you have a good documentation link, I'd be happy to add it as well.

Reply:
The TEP that I got just merged late last year is https://github.com/tektoncd/community/blob/master/teps/0035-document-tekton-position-around-policy-authentication-authorization.md

I have not added cycles yet to actually create the documentation in tektoncd/pipelines, but it is a goal of mine to get started this month.

This specific scenario was not listed in that TEP per se, but the TEP did not claim to provide the exhaustive list.  Just the starting list.  So as I start the doc, I will let the shipwright community know, and we can make sure I include this one when PR reviews start.